environment-dev.yml

```diff
@@ -1,12 +1,16 @@
 # conda env create -f environment-dev.yml
 name: birdhouse-deploy
 channels:
-  - conda-forge  # for bump2version
+  - conda-forge
+  - nodefaults  # to avoid installing the default packages that come with conda
 dependencies:
   - make
 
-  # make bump-install will install bump2version using pip, for user prefering pip
-  - bump2version
+  # make bump-install will install bump-my-version using pip, for user preferring pip
+  - bump-my-version
+
+  # for generating requirements.txt with hashes
+  - pip-tools ==7.5.0
```

Comment:
Could these be defined in a `pyproject.toml` as dev-dependencies so we get rid of all the various dependency files spread across the repo?

Reply:
The `conda` requirements won't work in a `pyproject.toml`, unfortunately.

If we want to use `pyproject.toml` we need to redefine `birdhouse-deploy` as an installable package. This means it would require metadata, a build backend, etc. This repo is an amalgam of setup scripts; I'm not sure if that makes the most sense.

Reply:
There would need to be something installed in the libraries. An example of a repo that installs only metadata is [`pavics-sdi`](https://github.com/Ouranosinc/pavics-sdi.).

We could technically do this with `pyproject.toml` using a backend that permits installing packages without any library files. It should be fine.